### PR TITLE
✨(frontend) adapt contract status with organization_signed_on

### PR DESCRIPTION
**Pull request by rlecellier**
Learner's training contract can have a new status where it's signed by the
learner but waiting the organization to counter-sign it.
There a small adaptation of our ContractStatus and
DashboardItemContract components

New state "waiting for organization to sign"
Capture d’écran du 2023-12-19 16-48-07

Modified state "Download your training contract"
Capture d’écran du 2023-12-18 17-15-03

### CODE REVIEW

File: src/frontend/js/components/ContractStatus/index.tsx [hunk 1]
```diff
@@ -4,34 +4,63 @@ import { Contract, ContractState } from 'types/Joanie';
 import { ContractHelper } from 'utils/ContractHelper';
 
 const messages = defineMessages({
-  signedOn: {
+  stutdenSignedOn: {
```
Comment (jbpenrath) [inline src/frontend/js/components/ContractStatus/index.tsx:7]:
```suggestion
  learnerSignedOn: {
```

File: src/frontend/js/components/ContractStatus/index.tsx [hunk 2]
```diff
@@ -4,34 +4,63 @@ import { Contract, ContractState } from 'types/Joanie';
...
     defaultMessage: 'You signed this training contract. Signed on {date}',
-    description: 'Label for the date of sign of a contract',
-    id: 'components.ContractStatus.signedOn',
+    description: 'Label for the date of sign of a training contract by the student',
+    id: 'components.ContractStatus.stutdenSignedOn',
```
Comment (jbpenrath) [inline src/frontend/js/components/ContractStatus/index.tsx:10]:
```suggestion
    description: 'Label for the date of sign of a training contract by the learner',
    id: 'components.ContractStatus.learnerSignedOn',
```

File: src/frontend/js/components/ContractStatus/index.tsx [hunk 3]
```diff
@@ -4,34 +4,63 @@ import { Contract, ContractState } from 'types/Joanie';
...
   },
-  waitingSignature: {
+  organizationSignedOn: {
+    defaultMessage: 'The organization have signed this training contract. Signed on {date}',
+    description: 'Label for the date of sign of a training contract by the organization',
+    id: 'components.ContractStatus.organizationSignedOn',
+  },
+  waitingStudentSignature: {
```
Comment (jbpenrath) [inline src/frontend/js/components/ContractStatus/index.tsx:17]:
```suggestion
  waitingLearnerSignature: {
```

File: src/frontend/js/components/ContractStatus/index.tsx [hunk 4]
```diff
@@ -4,34 +4,63 @@ import { Contract, ContractState } from 'types/Joanie';
...
     defaultMessage: 'You have to sign this training contract to access your training.',
-    description: 'Label displayed when a contract need to be signed',
+    description: 'Label displayed when a training contract need to be signed by the student',
```
Comment (jbpenrath) [inline src/frontend/js/components/ContractStatus/index.tsx:19]:
```suggestion
    description: 'Label displayed when a training contract need to be signed by the learner',
```

File: src/frontend/js/components/ContractStatus/index.tsx [hunk 5]
```diff
@@ -4,34 +4,63 @@ import { Contract, ContractState } from 'types/Joanie';
...
     id: 'components.ContractStatus.waitingSignature',
   },
+  waitingOrganization: {
+    defaultMessage:
+      'You cannot download your training contract until it have been sign but the organization.',
```
Comment (jbpenrath) [inline src/frontend/js/components/ContractStatus/index.tsx:24]:
```suggestion
      'You cannot download your training contract until it has been signed by the organization.',
```

File: src/frontend/js/widgets/Dashboard/components/DashboardItem/Contract/index.tsx [hunk 6]
```diff
@@ -35,13 +43,14 @@ export const DashboardItemContract = ({
...
               <ContractStatus contract={contract} />
             </span>
             <div>
-              {contract && contract.student_signed_on ? (
-                <DownloadContractButton contract={contract} />
-              ) : (
+              {contractState === ContractState.SIGNED && (
+                <DownloadContractButton contract={contract!} className="dashboard-item__button" />
+              )}
+              {contractState === ContractState.UNSIGNED && (
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Dashboard/components/DashboardItem/Contract/index.tsx:53]: This is already managed by the `SignContractButton` 🤔

**Review by jbpenrath — COMMENTED** [review]
Can you jut provide some screenshots please ?

File: src/frontend/js/widgets/Dashboard/components/DashboardItem/Contract/index.tsx (hunk 6 above)
Comment (rlecellier) [reply src/frontend/js/widgets/Dashboard/components/DashboardItem/Contract/index.tsx:53]: nice catch, i've fail my rebase and add back something that should not be there.

**Review by jbpenrath — APPROVED** [review]

**Pull request merged**